### DISCUSSION TITLE
Custom Range Templates

### DISCUSSION THREAD

**Discussion by unreified** [massif-press/compcon#2174] [Archive]
cross-posting from PilotNET:

I write third-party content for lancer, so I sometimes run into some weird usecases with the lancer framework that aren't necessarily needed normally. Usually it's stuff I can get around through other means by using stuff like profiles and tags. That said, there's some material that is just hardcoded into the system. Currently, I'm wondering the feasibility of custom Range Templates (Line/Blast/etc) in the same way as custom System/Weapon Tags, Statuses/Conditions, etc are supported.

I know I can submit a feature request (and I will, as long as I'm not misunderstanding something). But, since I imagine it'd be a very low priority task, would submitting a pull request to try and do it myself be a feasible thing if I have some hobbyist experience with git and javascript?

**Comment by msprijatelj** [comment]
New PRs for new features are welcome! I’d recommend the following:
1. Make a Feature Request anyway to provide detail on what precisely you’d like to add to CompCon
2. Check out the Contributors file for info on how to get setup with a local version of CompCon: https://github.com/massif-press/compcon/blob/master/CONTRIBUTORS.md

Right now, the range templates have dedicated icons in their compcon.ttf font file; my only hesitation at this point would be if there’s new icons required to add to the font file, and if so, how can we make them general enough for other homebrew to leverage?

**Reply by unreified** [reply]
Alright, will do! I've taken a look at the contributors file, and also done a cursory look through some of the relevant files (including the font file, to see if I could re-use things - maybe?).

That's a great point about generalization. As I have much less experience than you and am not great at estimating scope, do you think it would it be reasonable to implement the ability for .lcp files to pass in a font (hosted externally) and then have the method fetching icons use those as fallback options for compcon.tff? If so, this could be probably generalized to anything that requires its own icon (such as damage types/role tags/etc). That'd be more work, but it would mean that icon addition would no longer depend on maintenance of the compcon.ttf file and its dependencies.

I'll submit a proper issue for it after work today!
